Sprint-2/implement/querystring.js

```diff
@@ -1,16 +1,45 @@
 function parseQueryString(queryString) {
   const queryParams = {};
-  if (queryString.length === 0) {
+
+  // 〰️ Check for null, undefined, or non-string input
+  if (!queryString || typeof queryString !== "string") {
     return queryParams;
   }
+
+  // 〰️ Split the query string into key-value pairs using '&'
   const keyValuePairs = queryString.split("&");
 
   for (const pair of keyValuePairs) {
-    const [key, value] = pair.split("=");
+    // 〰️ Split each pair by the first '=' and keep remaining as the value
+    const [rawKey, ...rawValue] = pair.split("=");
+
+    // 〰️ Decode keys and values safely
+    // 〰️ Using decodeURIComponent() to convert percent-encoded strings into readable text.
+    // 〰️ Example: 'John%20Doe' -> 'John Doe', '5%25' -> '5%'
+    // 〰️ rawKey || "" ensures that if rawKey is undefined, null, or empty, it returns to an empty string.
+    const key = decodeURIComponent(rawKey || "");
+    const value = decodeURIComponent(rawValue.join("=") || "");
```

Comment:
you  demonstrates a solid understanding of value-key pair parsing. Maybe in order to enhance  test cases, you can also add dublicate test case  for keys i.e `"key=value1&key=value2"`  something like this.

Reply:
Thank you for your feedback @halilibrahimcelik! 

I've updated the implementation to handle duplicate keys by storing their values in an array. Additionally, I’ve added a test case to validate this behavior, ensuring that query strings like key=value1&key=value2 are correctly parsed into { key: ["value1", "value2"] }. I hadn’t considered this, but your suggestion helped improve my code. Thanks again!

Let me know if there’s anything else I can improve. 🙌

Reply:
I'm glad to hear that! It’s a niche case, and in a real-world scenario, it’s unlikely to have two different values with the same key. However, it’s a good challenge, keep up the good work! Cheers 🎉

Reply:
Thank you! 🤗